gitlabform/gitlab/python_gitlab.py

```diff
@@ -0,0 +1,61 @@
+import functools
+
+from gitlab import Gitlab, GitlabGetError
+from gitlab.base import RESTObject, RESTObjectList
+from gitlab.v4.objects import Group
+
+
+# Extends the python-gitlab class to add convenience wrappers for common functionality used within gitlabform
+class PythonGitlab(Gitlab):
+    def get_user_id(self, username) -> int:
+        user = self.get_user_by_username(username)
+        return user.id
+
+    def get_group_id(self, groupname) -> int:
+        groups = self.get_group_by_groupname(groupname)
+        return groups.id
+
+    def get_group_by_name(self, groupname: str) -> Group:
+        group = self.get_group_by_groupname(groupname)
+        return self.groups.get(group.id)
+
+        #  Uses "LIST" to get a group by groupname, to get the full Group object, call get using the group's id
+
+    @functools.lru_cache()
+    def get_group_by_groupname(self, groupname: str) -> RESTObject:
+        groups = self.groups.list(search=groupname)
```

Comment:
I am sorry, but this is a wrong way to get a group by its exact name. 😞 

It's counterintuitive, I know, but In a big enough GitLab instance such a search will return many groups and not necessarily the expected, right one as the first.

Just look at what gitlab.com returns for "foo":
```
>>> [group.name for group in gl.groups.list(search="foo")]
['2019_SeeFood_Hooli', '2023-06-01 Project Food Ordering', '346FoodTrucks', '6to11', '80foods', 'AF v6.0.0', 'AG Bezirkswebsite', 'API - BackEnd', 'AWS Deployment', 'Action Eva Flotter', 'Adpro B9 Food Sharing', 'Ajinomoto Foods', 'Ajinomoto Foods', 'Ajinomoto Foods', 'Ajinomoto Foods', 'Ajinomoto Foods NA', 'Aljaya Food', 'Angular FrontEnd', 'Anything Football', 'Apache Foot and Ankle Specialists']
```
The expected https://gitlab.com/foo group is not even in these first 20 results!

The right way to get it would be
```
>>> gl.groups.get('foo')
<Group id:44565 name:foo>
```

(I based this suggestion on [this answer for getting a specific project](https://github.com/python-gitlab/python-gitlab/issues/710#issuecomment-466623644).)

Could you please correct this method accordingly, @TimKnight-DWP?

Reply:
@gdubicki perfect thanks